### PR TITLE
fix: after proxy-mirror plugin mirroring request, the host in the mir…

### LINKED ISSUE #2617: bug(proxy-mirror): after proxy-mirror plugin mirroring request, the host in the mirroring request has changed
### Issue description

When I only used the `proxy-mirror` plugin, I found that the `host` in the mirroring request was not the host in the original request, but the host configured in the `proxy-mirror` plugin.

This is my test case in `proxy-mirror.t` :

1.Add 1986 port service, used to print mirroring request information in the log.

```
add_block_preprocessor(sub {
    my ($block) = @_;

    my $http_config = $block->http_config // <<_EOC_;

    server {
        listen 1986;
        server_tokens off;

        location / {
            content_by_lua_block {
                local core = require("apisix.core")
                local headers_tab = ngx.req.get_headers()                
                core.log.warn(ngx.var.request)
                for k, v in pairs(headers_tab) do                    
                    core.log.warn( "k: ", k, " v: ", v)
                end                
            }
        }
    }
_EOC_

    $block->set_value("http_config", $http_config);
});
```

2.Configure `proxy-mirror` plugin information

```
=== TEST 1: sanity check (normal case)
--- config
       location /t {
           content_by_lua_block {
               local t = require("lib.test_admin").test
               local code, body = t('/apisix/admin/routes/1',
                    ngx.HTTP_PUT,
                    [[{
                        "plugins": {
                            "proxy-mirror": {
                               "host": "http://127.0.0.1:1986"
                            }
                        },
                        "upstream": {
                            "nodes": {
                                "127.0.0.1:1980": 1
                            },
                            "type": "roundrobin"
                        },
                        "uri": "/uri"
                   }]]
                   )

               if code >= 300 then
                   ngx.status = code
               end
               ngx.say(body)
           }
        }
--- request
GET /t
--- error_code: 200
--- response_body
passed
--- no_error_log
[error]
```

3.Send a request to test the plugin.

```
=== TEST 9: 
--- request
GET /uri
--- error_code: 200
--- more_headers
host: 127.0.0.1
api-key: hello
pageSize: 11
name: jake
--- response_body
uri: /uri
host: 127.0.0.1
pagesize: 11
x-real-ip: 127.0.0.1
name: jake
api-key: hello
--- no_error_log
[error]
```

4.View the printed log, host is the host configured in the proxy-mirror plugin :

image

### PR DESCRIPTION
**Pull request by Firstsawyou**
…roring request has changed.

fix #2617

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: apisix/cli/ngx_tpl.lua [hunk 1]
```diff
@@ -498,6 +498,7 @@ http {
                 return 200;
             }
 
+            proxy_set_header   Host       $upstream_host;
```
Comment (spacewander) [inline apisix/cli/ngx_tpl.lua:501]: The spaces between words here is weird.

Comment (Firstsawyou) [reply apisix/cli/ngx_tpl.lua:501]: Consider keeping it aligned with the line below. What do you think? Do we need to adjust?

Comment (spacewander) [reply apisix/cli/ngx_tpl.lua:501]: Would be better to be aligned. It looks tidy.

**Review by gxthrj — APPROVED** [review]

File: apisix/cli/ngx_tpl.lua (hunk 1 above)
Comment (membphis) [reply apisix/cli/ngx_tpl.lua:501]: > Would be better to be aligned. It looks tidy.

agree

**Review by spacewander — APPROVED** [review]

**Pull request merged**